Sources/Swift/Integrations/SessionReplay/SentryReplayOptions.swift

```diff
@@ -158,5 +162,20 @@ public class SentryReplayOptions: NSObject, SentryRedactOptions {
         self.unmaskedViewClasses = ((dictionary["unmaskedViewClasses"] as? NSArray) ?? []).compactMap({ element in
             NSClassFromString((element as? String) ?? "")
         })
+        if let quality = (dictionary["quality"] as? String) {
+            self.quality = SentryReplayQuality.fromName(quality)
+        }
```

Comment:
m: This is only for hybrid SDKs, we dont need the string to int conversion. 
Let's also use int in the dictionary. Use a Int "enum" in the RN side too and all good.

Reply:
We serialize the data to pass them to native and we can't use the enum in JS/Dart...

And the number without the enum is not clear.

Reply:
> And the number without the enum is not clear

Does it need to be clear? Should this dictionary be readable or it's just a way to pass information from RN to native?

What is the data type of this property in the RN options? Can you create a converter from that data type to int and pass the int?

Reply:
It will be a string `low`, `medium` and `high`. 

Technically the dict can be a number, it doesn't have to be readable when passing thru layers, only serializable.

---

By having the conversion here all the hybrids could use it.

Reply:
I can split the dict and the string to enum int conversion.

https://github.com/getsentry/sentry-cocoa/pull/4495/commits/6df0f60f0f9d2b5c651e1897fae1823c9a879ec2

Reply:
Using enum (and by extend numbers) is less prone to errors. Also there is the performance aspect of conversions.

I would suggest you to not use string in the RN property, JS dont have enum but you could use some const int values to minimize user mistakes. 

But I will leave it up to you. We can keep this as it is.

Reply:
Yeah, sadly JS/TS enums are bad. Can const int is a solution, but hard to discover for JS devs, but that's out of scope for this PR. 

I was mainly interested in the `why numbers`. Thank you for the details.